horovod/common/operations.cc

```diff
@@ -205,17 +205,17 @@ OperationManager* CreateOperationManager(HorovodGlobalState& state) {
   broadcast_ops.push_back(
       std::shared_ptr<BroadcastOp>(new MPIBroadcast(&mpi_context, &state)));
 
+  std::shared_ptr<JoinOp> join_op(new JoinOp(&state));
   std::shared_ptr<ErrorOp> error_op(new ErrorOp(&state));
 
-  return new OperationManager(&state.param_manager, allreduce_ops,
-                              allgather_ops, broadcast_ops, error_op);
+  return new OperationManager(&state.param_manager, allreduce_ops, allgather_ops, broadcast_ops, join_op, error_op);
 }
 
 // Store the Request for a name, and return whether the total count of
 // Requests for that tensor is now equal to the MPI size (and thus we are
 // ready to reduce the tensor).
 bool IncrementTensorCount(std::unique_ptr<MessageTable>& message_table,
-                          const Request& msg, int mpi_size) {
+                          const Request& msg, int mpi_size, int joined_size = 0) {
```

Comment:
NIT: should it be uint32_t instead of int?

Reply:
This should be the same type as `HorovodGlobalState.size`, which is `int`. I'll change `HorovodGlobalState.joined_size` to be `int`.
BTW, this PR is not exactly ready for review at the moment...